versioned_docs/version-2.0.0/running-keploy/configuration-file.md

```diff
@@ -44,65 +56,259 @@ test:
+
+- **`path`**: Specifies the path for which the test should be recorded. It defines the URL path of the API endpoint.
+
+- **`urlMethods`**: Specifies the HTTP methods for which the test should be recorded. It allows you to focus on specific HTTP methods like GET, POST, etc.
+
+- **`headers`**: Specifies headers and their values for which the test should be recorded. It enables you to filter tests based on specific headers.
+
+- **`host`**: Specifies the host for which the test should be recorded. It defines the domain or IP address of the API server.
+
+#### Using Test Filters Together or Independently
+ You can use the **`path`**, **`urlMethods`**, **`headers`**, and **`host`** filters together or independently based on your testing scenarios. This flexibility allows you to precisely define the conditions under which tests are recorded.
+
+### Stubs Section
+
+The stubs section in the Keploy-config file allows you to define parameters for bypassing and mocking API calls.
+
+- **`filters`**: Filters to define stubs for specific paths, ports, and hosts.
+
+  Example:
+
+  ```yaml
+  stubs:
+    filters:
+      - path: "/user/app"
+        port: 8080
+      - port: 8081
+      - host: "dc.services.visualstudio.com"
+      - port: 8081
+        host: "dc.services.visualstudio.com"
+        path: "/user/app"
+    ```
+  The stubs section provides a way to bypass and mock API calls during testing. The filters subsection allows you to define specific conditions for applying stubs, such as path, port, and host. You can use these filters together or independently based on your testing scenarios.
+
+  - **`path`**: Specifies the path for which the stub should be applied. It defines the URL path of the API endpoint.
+
+  - **`port`**: Specifies the port for which the stub should be applied. It defines the network port on which the API call is made.
+
+  - **`host`**: Specifies the host for which the stub should be applied. It defines the domain or IP address of the API server.
+
+  In the provided example:
+
+  - The first stub applies to the path "/user/app" and the port 8080.
+  - The second stub applies to the port 8081.
+  - The third stub applies to the host "dc.services.visual
+## Advanced Noise Filtering:
+
+Earlier the only way to add the [noisy fields](http://keploy.io/docs/concepts/general-glossary/#3-noisy-field) was by modifying individual test file (testcase level). Now, With the introduction of config file, users can add the noisy fields at test-set and global level through config file itself.
+
+### Global Noise
+
+The Global Noise section is used to define parameters that are globally ignored for all API calls during testing. It enables you to filter out consistent noise, ensuring a cleaner evaluation of responses.
```

Comment:
replace "The Global Noise section" to "The `global` subsection of `globalNoise`"